### PR TITLE
BUG: MaskedArray _optinfo dictionary is not updated when calling __eq__/__ne__

### PR DESCRIPTION
**Pull request by schaefed**
Replaces PR #9279. In order to follow the contribution guidelines (I think) I had to open a new one...

### CODE REVIEW

File: numpy/ma/tests/test_core.py [hunk 1]
```diff
@@ -691,6 +691,27 @@ def test_optinfo_propagation(self):
...
+        assert_equal(a._optinfo["key"], (a * 2)._optinfo["key"])
+        assert_equal(a._optinfo["key"], (a / 2)._optinfo["key"])
+        assert_equal(a._optinfo["key"], a[:2]._optinfo["key"])
+        assert_equal(a._optinfo["key"], a[[0,0,2]]._optinfo["key"])
+        assert_equal(a._optinfo["key"], np.exp(a)._optinfo["key"])
+        assert_equal(a._optinfo["key"], np.abs(a)._optinfo["key"])
+        assert_equal(a._optinfo["key"], array(a, copy=True)._optinfo["key"])
+        assert_equal(a._optinfo["key"], np.zeros_like(a)._optinfo["key"])
+
+
```
Comment (eric-wieser) [inline numpy/ma/tests/test_core.py]: Should only be one line between methods - I don't think we make an exception for tests?

**Review by eric-wieser — APPROVED** [review]
One minor style nit, but fix looks fine, and tests are thorough

File: numpy/ma/tests/test_core.py (hunk 1 above)
Comment (charris) [inline numpy/ma/tests/test_core.py]: We don't, although I expect violations slip by. Might as well get it right in the first place.

**Comment by schaefed** [comment]
I removed the excessive empty lines

**Comment by eric-wieser** [comment]
Thanks @schaefed!

Try to use the `BUG: ` prefix on the commit message itself in future, not just the PR name - not a big deal here though - i just did a squash and merge

**Pull request merged**